frontend/src/store/FamLoginUserState.ts

```diff
@@ -215,66 +228,93 @@ const getCachedAppRolesForDelegatedAdmin = (
             : 1;
     });
 };
+//--------- get my permissions
+const getMyFamAdminPermission = (access: FamAuthGrantDto): IMyPermission[] => {
+    const famGrant = access.grants.find(
+        (grant: FamGrantDetailDto) =>
+            grant.application.name === FAM_APPLICATION_NAME
+    );
+    if (!famGrant) return [];
 
-const getMyAdminPermission = () => {
-    let myPermissions: any = [];
-    getUserAccess()?.forEach((item) => {
-        if (item.auth_key === AdminRoleAuthGroup.FamAdmin) {
-            const famGrant = item.grants.find((grant) => {
-                return grant.application.name === FAM_APPLICATION_NAME;
-            });
-            myPermissions.push({
-                application: famGrant?.application.description,
-                env: famGrant?.application.env,
-                role: 'Admin',
-            });
-        }
+    return [
+        {
+            application: famGrant.application.description,
+            env: famGrant.application.env,
+            role: 'Admin',
+        },
+    ];
+};
 
-        if (item.auth_key === AdminRoleAuthGroup.AppAdmin) {
-            item.grants.forEach((grant) => {
-                myPermissions.push({
-                    role: 'Admin',
+const getMyAppAdminPermission = (access: FamAuthGrantDto): IMyPermission[] => {
```

Comment:
For those internal methods: "getMyFamAdminPermission", "getMyAppAdminPermission" and "getMyDelegatedAdminPermission", can we simply placed them above the "// --- export"?
Thanks.